monkestation/code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -0,0 +1,276 @@
+
+/obj/item/clothing/suit/armor/reactive/glacial/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return FALSE
+	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The walter fabricator is still recharging!</span>")
+			return FALSE
+		playsound(get_turf(src),'sound/effects/empulse.ogg', 100, 1)
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out an icy blast!</span>")
+		var/turf/owner_turf = get_turf(owner)
+		for(var/mob/living/carbon/target_atom as mob in oviewers(7, owner_turf))
+			var/freezing_power = 50
+			freezing_power -= (5*get_dist(owner_turf,target_atom))
+			target_atom.adjust_bodytemperature(-freezing_power) //freezes less from further away
+
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+		return TRUE
+
+/obj/item/clothing/suit/armor/reactive/glacial/emp_act()
+	for(var/mob/living/carbon/target_atom as mob in viewers(7, get_turf(src)))
+		var/freezing_power = rand(1,100) //:) I love EMPS :)
+		if(freezing_power<30)
+			src.visible_message("<span class='danger'>[src] malfunctions, letting out an cold breeze.</span>")
+		else if(freezing_power<60)
+			src.visible_message("<span class='danger'>[src] malfunctions, chilling the air around it.</span>")
+		else
+			src.visible_message("<span class='danger'>[src] malfunctions, forming ice in the air around you.</span>")
+		freezing_power -= (5*get_dist(get_turf(src),target_atom))
+		target_atom.adjust_bodytemperature(-freezing_power)
+	return
+
+//Monkey Armour
+/obj/item/clothing/suit/armor/reactive/primal
+	name = "reactive primal armor"
+	desc = "An experimental suit of armor that echoes the screeches of past monkeys."
+	reactivearmor_cooldown_duration = 3 MINUTES //Big Cooldown
+	hit_reaction_chance = 10 //Low monkey chance
+
+/obj/item/clothing/suit/armor/reactive/primal/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return FALSE
+	if(prob(hit_reaction_chance))
+		if(world.time < reactivearmor_cooldown)
+			owner.visible_message("<span class='danger'>The monkey generator is still recharging!</span>")
+			return FALSE
+		playsound(get_turf(src),'sound/creatures/monkey/monkey_screech_1.ogg', 100, 1)
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], and screeches with the voices of a million monkeys!</span>")
+		return_to_monkey(owner)
+
+		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+		return TRUE
+
+/obj/item/clothing/suit/armor/reactive/primal/proc/return_to_monkey(mob/user)
+	if(!ishuman(user))
+		return
+	var/mob/living/simple_animal/pet/new_gorilla = new /mob/living/simple_animal/hostile/gorilla/rabid(get_turf(user))
```

Comment:
Will do a more in-depth review at a later time, but absolutely do not use the rabid subtype.
Anything related to the monkey gamemode is extremely busted.

Reply:
ok 👍 will change it to a normal gorilla if that'd be better. otherwise I could just make it a normal Monkey